src/interpreter/Interpreter.cpp

```diff
@@ -1803,10 +1806,76 @@ ByteCodeStackOffset* Interpreter::interpret(ExecutionState& state,
             Trap::throwException(state, "null i31 reference");
         }
         writeValue<int32_t>(bp, code->dstOffset(), Value::getI31UValue(ptr));
+
         ADD_PROGRAM_COUNTER(I31GetU);
         NEXT_INSTRUCTION();
     }
 
+    DEFINE_OPCODE(StructNew)
+        :
+    {
+        StructNew* code = (StructNew*)programCounter;
+
+        GCStruct* result = GCStruct::structNew(code->typeInfo(), code->dataOffsets(), bp);
```

Comment:
Who roots(prevent freeing from GC) result pointer?
you can write the comment. (since the gc spec is minor for wasm)

Reply:
Honestly, this part is probably needs a lot of improvements later.

The result goes to the stack, unless a memory is allocated by the executor:
https://github.com/Samsung/walrus/blob/main/src/Walrus.h#L352
This macro needs to be improved somehow, but it looks a bit complex.

The result can copied into a table (tables are not garbage collected at this moment), and a wasm value
https://github.com/Samsung/walrus/blob/main/src/api/wasm.h#L191

Is there a hard reference counting in GC (a memory area cannot be freed until its reference counter is zero)? So a wasm_valtype_t could increase/decrease this number to prevent freeing a reference.

The stack is good enough for the current tests. Shall I add a TODO comment?

Reply:
IMO this way is not sufficient
we sometimes lost GCStruct* when Garbage collecting
since stack can be allocated by libc's malloc
I think we can table as root-space for gc
you can malloc table memory from gc and mark from callback where gc struct located
or
you can implement your idea

Reply:
I have used `GC_MALLOC_UNCOLLECTABLE / GC_FREE` to create a large stack space or table data. Is this ok? The description in gc.h says these pointer arrays are never collectible.